web/packages/shared/utils/highbar.ts

```diff
@@ -60,33 +60,55 @@ export function mergeDeep(target: MergeTarget, ...sources: Array<MergeTarget>) {
   return mergeDeep(target, ...sources);
 }
 
-type CompareArray = Array<Record<string, unknown>>;
+/** Recursively freezes an object and its contents. */
+export function freezeDeep<T extends object>(obj: T): T {
+  // Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze
+  // Any copyright is dedicated to the Public Domain:
```

Comment:
IANAL and also not an authoritative comment*:

I do not think we need to add a copyright note here. I get that the way to recursively check for deep equality is taken as a reference from the MDN but the `Object.freeze` is part of the core JS.

Source reference link is good though.